### DISCUSSION TITLE
Where is the java-admin-client server implementation?

### DISCUSSION THREAD

**Discussion by JudeNiroshan** [keycloak/keycloak#22000] [Q&A] [answered]
I cannot find the relevant Quarkus Resource classes with annotated REST endpoints. At debug runtime, I only see proxy classes. I could not find documentation about this. Can someone point it to me?

**Comment by vmuzikar** [comment]
You can find the sources for the admin client lib [here](https://github.com/keycloak/keycloak/tree/main/integration/admin-client-jee/src/main/java/org/keycloak/admin/client)

**Reply by JudeNiroshan** [reply]
Maybe I didn't ask the question properly. What I'm asking is, E.g. where is the implementation for this interface? https://github.com/keycloak/keycloak/blob/8ad6bc49a3903cfefcc26c251890c35de8fa050e/integration/admin-client-jee/src/main/java/org/keycloak/admin/client/resource/ClientResource.java#L46

**Reply by danielFesenmeyer** [reply, marked as answer]
There are no implementations - it's magic based on annotation-processing and proxies ;)
To be more specific, Resteasy proxies are used to handle the REST calls, see https://docs.jboss.org/resteasy/docs/6.2.0.Final/userguide/html/RESTEasy_Client_Framework.html#proxies.
